lib/index.js

```diff
@@ -46,187 +43,103 @@ Ramen.prototype.initialize = function() {
+  }
 
-Ramen.prototype.identify = function(identify) {
-  privateGo({ integration: this, identify: identify });
-};
+  var props = group.properties();
+  props = alias(props, { createdAt: 'created' });
+  props = alias(props, { created: 'created_at' });
+  var id = group.groupId();
+  if (id) props.id = id;
 
-Ramen.prototype.group = function(group) {
-  privateGo({ integration: this, group: group });
+  window._ramen.group(props);
 };
 
-Ramen.prototype.page = function(page) {
-  privateGo({ integration: this, page: page });
+Ramen.prototype.page = function() {
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
+
+  window._ramen.page();
 };
 
 Ramen.prototype.track = function(track) {
-  privateGo({ integration: this, track: track });
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
+
+  var properties = track.properties();
+  properties = convertDates(properties, function(date) { return Math.floor(date / 1000); });
+  window._ramen.track(track.event(), properties);
 };
 
-function privateGo(options) {
-  var integration = options.integration;
-  var ramenSettings = {};
-  var property;
-  var identify;
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Identify} identify
+ */
+
+Ramen.prototype.identify = function(identify) {
+  var user;
   var traits;
   var opts;
-  var user_opts;
-  var created;
-  var name;
-  var id;
-  var email;
-  var group;
-
-  if (options.identify) {
-    identify = options.identify;
-    traits = identify.traits();
-    opts = identify.options(integration.name);
-    user_opts = opts.user;
-    created = identify.created();
-    name = identify.name();
-    id = identify.uid();
-    email = identify.email();
-    group = integration.analytics.group();
-  } else {
-    traits = {};
-    group = {};
-    opts = {};
-  }
 
-  email = email || integration.analytics.user().traits().email;
-  id = id || integration.analytics.user().id();
+  this.identified = true;
 
-  // We need email addresses
-  if (!email) {
+  if (identify.anonymous) {
+    window._ramen.identify();
     return;
   }
 
-  del(traits, 'email');
-  del(traits, 'name');
-
-  if (traits.company !== null && !is.object(traits.company)) {
-    delete traits.company;
-  }
-
-  if (traits.company && typeof group.traits === 'function') {
-    defaults(traits.company, group.traits());
-  }
-
-  ramenSettings.organization_id = integration.options.organization_id;
-  ramenSettings.user = {
-    id: id,
-    email: email
-  };
+  traits = identify.traits();
+  opts = identify.options(this.name);
 
-  if (name) {
-    ramenSettings.user.name = name;
-  }
+  // Setup the basic `user` attributes: id, email, created_at, and name
+  // `null` values are OK. Ramen will ignore them.
 
-  if (created) {
-    ramenSettings.user.created_at = Math.round(created / 1000);
-    del(traits, 'created_at');
-    del(traits, 'created');
-    del(traits, 'createdAt');
-  }
+  user = { id: identify.userId() };
```

Comment:
Note that if someone calls `analytics.identify({ someTrait: 'hello' });` aka without any `userId` which can definitely happen, this would be setting `user = { id: undefined }`